tests/special_coverage/test_spec_cov_search.py

```diff
@@ -2,16 +2,75 @@
 
 from django.utils import timezone
 
-from bulbs.content.filters import FeatureTypes
+from bulbs.campaigns.models import Campaign
 from bulbs.content.models import Content, FeatureType, Tag
-from bulbs.special_coverage.search import second_slot_query_generator
+from bulbs.special_coverage.models import SpecialCoverage
+from bulbs.special_coverage.search import ReadingListIterator, SearchParty
 from bulbs.utils.test import BaseIndexableTestCase, make_content
 
 
+class ReadingListIteratorTestCase(BaseIndexableTestCase):
+
+    def setUp(self):
+        super(ReadingListIteratorTestCase, self).setUp()
+        self.now = timezone.now()
+        self.feature_type1 = FeatureType.objects.create(name="Arguments")
+        self.feature_type2 = FeatureType.objects.create(name="Falling")
+        for i in range(20):
+            Content.objects.create(
+                title="content %s" % i,
+                published=self.now - timezone.timedelta(hours=i),
+                feature_type=self.feature_type1
+            )
+            Content.objects.create(
+                title="article %s" % i,
+                published=self.now - timezone.timedelta(hours=i),
+                feature_type=self.feature_type2
+            )
+        Content.search_objects.refresh()
+
+    def test_register_queryset_default_first(self):
+        queryset = Content.search_objects.search(feature_types=[self.feature_type1.slug])
+        reading_list = ReadingListIterator()
+        # First queryset should be default until explicitly state otherwise.
+        reading_list.register_queryset(queryset)
+        self.assertEqual(reading_list.default_queryset, queryset)
+        out = [obj for obj in reading_list]
+        for obj in queryset.full():
```

Comment:
the problem with this approach is that if queryset ever returns unexpected results (maybe ES mapping changes, ES request silently fails and no results returned, etc.), the testing assertions will never run. You could add a length check, but better to just be explicit and compare with an array of the exact expected results:

`self.assertEqual(out, [self.content[0], self.content[10], ....])`